zipkin-server/src/main/java/zipkin/server/ZipkinServerConfiguration.java

```diff
@@ -46,8 +48,14 @@ CollectorSampler traceIdSampler(@Value("${zipkin.collector.sample-rate:1.0}") fl
 
   @Bean
   @ConditionalOnMissingBean(CollectorMetrics.class)
-  CollectorMetrics metrics(CounterService counterService, GaugeService gaugeService) {
-    return new ActuateCollectorMetrics(counterService, gaugeService);
+  CollectorMetrics metrics(Optional<CounterBuffers> counterBuffers, Optional<GaugeBuffers> gaugeBuffers) {
```

Comment:
oh, makes sense. I thought you meant whether ActuateCollectorMetrics gets instantiated at all with default configs. Unit test is probably OK, but might not cover use cases with multiple Collectors present. Alternatively we can make ActuateCollectorMetrics completely independent and always instantiate CounterBuffers/GaugeBuffers in the ActuateCollectorMetrics's constructor.